File: 1-100/66.plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int nCarry = 1;
        for(auto itr = digits.rbegin(); itr != digits.rend(); ++itr)
        {
            int sum = *itr + nCarry;
            *itr = sum % 10;
            nCarry = sum / 10;
        }

        if (nCarry == 1)
        {
            digits.insert(digits.begin(), nCarry);
        }

        return digits;
    }
};

File: 1-100/66.plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        nCarry = 1
        nIdx = len(digits) - 1
        while nIdx >= 0:
            digits[nIdx] += 1
            if digits[nIdx] > 9:
                digits[nIdx] -= 10
                nIdx -= 1
            else:
                return digits
        
        digits.insert(0, 1)

        return digits
